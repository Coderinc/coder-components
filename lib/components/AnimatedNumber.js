import React from 'react';
import raf from 'raf';

class AnimatedNumber extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      currentNumber: 0,
    };

    this.confirmSixtyFPS = this.confirmSixtyFPS.bind(this);
    this.prepTween = this.prepTween.bind(this);
    this.endTween = this.endTween.bind(this);
    this.tweenValue = this.tweenValue.bind(this);
  }

  componentDidMount() {
    this.prepTween();
  }

  componentDidUpdate(prevProps) {
    if (this.state.currentNumber === this.props.number) {return;}
    if (prevProps.number === this.props.number) {return;}
    if (this.tweenHandler) {this.endTween();}

    this.prepTween();
  }

  confirmSixtyFPS(timestamp) {
    const { lastTimestamp } = this.state;
    return !lastTimestamp || timestamp - lastTimestamp > 16;
  }

  prepTween() {
    this.tweenHandler = raf(timestamp => {
      this.tweenValue(timestamp, true);
    });
  }

  endTween() {
    raf.cancel(this.tweenHandler);
  }

  tweenValue(timestamp, start = false) {
    if (!this.confirmSixtyFPS(timestamp)) {
      this.tweenHandler = raf(this.tweenValue);
      return;
    }

    const { number, duration } = this.props;
    const { currentNumber } = this.state;
    const startTime = start ? timestamp : this.state.startTime;
    const fromNumber = start ? currentNumber : this.state.fromNumber;

    let newNumber;
    if (timestamp - startTime >= duration) {
      newNumber = number;
    } else {
      newNumber =
        fromNumber +
        (number - fromNumber) * ((timestamp - startTime) / duration);
    }

    if (newNumber === number) {
      this.endTween();
      this.setState({
        currentNumber: number,
      });
      return;
    }

    this.setState({
      currentNumber: newNumber,
      startTime: startTime || timestamp,
      fromNumber,
      lastTimestamp: timestamp,
    });
    this.tweenHandler = raf(this.tweenValue);
  }

  render() {
    const displayValue = this.props.noDecimal
      ? Math.round(this.state.currentNumber)
      : this.state.currentNumber;

    const C = this.props.component;
    const formattedValue = this.props.format
      ? this.props.format(displayValue)
      : displayValue;

    if (C) {
      return <C>{formattedValue}</C>;
    }

    return <span>{formattedValue}</span>;
  }
}

export default AnimatedNumber;
