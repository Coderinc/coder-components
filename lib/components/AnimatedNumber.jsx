import React from 'react';
import PropTypes from 'prop-types';
import raf from 'raf';

class AnimatedNumber extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      currentNumber: 0,
    };

    this.confirmSixtyFPS = this.confirmSixtyFPS.bind(this);
    this.prepTween = this.prepTween.bind(this);
    this.endTween = this.endTween.bind(this);
    this.tweenValue = this.tweenValue.bind(this);
  }

  componentDidMount() {
    this.prepTween();
  }

  componentDidUpdate(prevProps) {
    const { currentNumber } = this.state;
    const { number } = this.props;
    if (currentNumber === number) {
      return;
    }
    if (prevProps.number === number) {
      return;
    }
    if (this.tweenHandler) {
      this.endTween();
    }

    this.prepTween();
  }

  confirmSixtyFPS(timestamp) {
    const { lastTimestamp } = this.state;
    return !lastTimestamp || timestamp - lastTimestamp > 16;
  }

  prepTween() {
    this.tweenHandler = raf((timestamp) => {
      this.tweenValue(timestamp, true);
    });
  }

  endTween() {
    raf.cancel(this.tweenHandler);
  }

  tweenValue(timestamp, start = false) {
    if (!this.confirmSixtyFPS(timestamp)) {
      this.tweenHandler = raf(this.tweenValue);
      return;
    }

    const { number, duration } = this.props;
    const { currentNumber, startTime, fromNumber } = this.state;
    const validatedStartTime = start ? timestamp : startTime;
    const validatedFromNumber = start ? currentNumber : fromNumber;

    let newNumber;
    if (timestamp - startTime >= duration) {
      newNumber = number;
    } else {
      newNumber = fromNumber + (number - fromNumber) * ((timestamp - startTime) / duration);
    }

    if (newNumber === number) {
      this.endTween();
      this.setState({
        currentNumber: number,
      });
      return;
    }

    this.setState({
      currentNumber: newNumber,
      startTime: validatedStartTime || timestamp,
      fromNumber: validatedFromNumber,
      lastTimestamp: timestamp,
    });
    this.tweenHandler = raf(this.tweenValue);
  }

  render() {
    const { noDecimal, format, component } = this.props;
    const { currentNumber } = this.state;
    const displayValue = noDecimal ? Math.round(currentNumber) : currentNumber;

    const C = component;
    const formattedValue = format ? format(displayValue) : displayValue;

    if (C) {
      return (
        <C>
          {formattedValue}
        </C>
      );
    }

    return (
      <span>
        {formattedValue}
      </span>
    );
  }
}

AnimatedNumber.propTypes = {
  number: PropTypes.number,
  duration: PropTypes.number,
  noDecimal: PropTypes.bool,
  format: PropTypes.func,
  component: PropTypes.func,
};

AnimatedNumber.defaultProps = {
  number: 0,
  duration: 250,
  noDecimal: false,
  format: undefined,
  component: undefined,
};

export default AnimatedNumber;
