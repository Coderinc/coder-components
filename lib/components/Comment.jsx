import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import moment from 'moment';
import Avatar from '../elements/Avatar';
import Paragraph from '../elements/Paragraph';

const CommentWrapper = styled.div`
  display: flex;
  width: 100%;
  padding: 1rem;
`;

const StyledAvatar = styled(Avatar)`
  margin-top: 0.375rem;
`;

const Text = Paragraph.extend`
  margin: 0 1rem 0 1rem;
  flex: 1;
`;

const Time = Paragraph.extend`
  color: ${({ theme }) => theme.offsetFont};
  font-size: 0.75rem;
  max-width: 5rem;
  text-align: right;
`;

const Comment = ({ user, comment, createdAt }) => (
  <CommentWrapper>
    <StyledAvatar {...user} large />
    <Text>
      {comment}
    </Text>
    <Time>
      {moment(createdAt).calendar()}
    </Time>
  </CommentWrapper>
);

Comment.propTypes = {
  user: PropTypes.object.isRequired,
  comment: PropTypes.string.isRequired,
  createdAt: PropTypes.oneOf([PropTypes.object, PropTypes.string]).isRequired,
};

export default Comment;
