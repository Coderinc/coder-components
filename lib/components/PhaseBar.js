import React from 'react';
import styled from 'styled-components';
import Paragraph from '../elements/Paragraph';
import Chip from '../elements/Chip';

const BarContainer = styled.div`
  background-color: ${props =>
    props.active
      ? props.theme.activeBackground
      : props.theme.primaryBackground};
  color: ${props => props.theme.primaryFont};

  display: flex;
  align-items: center;

  box-sizing: border-box;
  width: 100%;
  padding: 0.5rem;
  border-radius: 0.25rem;

  &::before {
    font-size: 1.5rem;
    font-family: coder-icons;
    font-weight: normal;
    content: '\f108';
    margin-right: 0.5rem;
  }
`;

const Fill = styled.div`
  flex: 1;
`;

export const PhaseBar = ({ label, phase, ...props }) => (
  <BarContainer {...props}>
    <Paragraph>{label}</Paragraph>
    <Fill />
    <Chip label={phase} />
  </BarContainer>
);

export default PhaseBar;
