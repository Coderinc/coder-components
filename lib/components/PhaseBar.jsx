import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Paragraph from '../elements/Paragraph';
import Chip from '../elements/Chip';

const BarContainer = styled.div`
  background-color: ${({ active, theme }) => (active ? theme.activeBackground : theme.offsetBackground)};
  color: ${({ theme }) => theme.primaryFont};

  display: flex;
  align-items: center;

  box-sizing: border-box;
  width: 100%;
  margin: 0.25rem 0;
  padding: 0.5rem;
  border-radius: 0.25rem;

  &::before {
    font-size: 1.5rem;
    font-family: coder-icons;
    font-weight: normal;
    content: '\f108';
    margin-right: 0.5rem;
  }
`;

const Fill = styled.div`
  flex: 1;
`;

const PhaseBar = ({
  label, phase, value, onClick, ...props
}) => (
  <BarContainer {...props} onClick={onClick ? () => onClick(value) : undefined}>
    <Paragraph>
      {label}
    </Paragraph>
    <Fill />
    <Chip label={phase} />
  </BarContainer>
);

PhaseBar.propTypes = {
  label: PropTypes.string.isRequired,
  phase: PropTypes.string.isRequired,
  // eslint-disable-next-line
  value: PropTypes.any,
  onClick: PropTypes.func,
};

PhaseBar.defaultProps = {
  value: 0,
  onClick: undefined,
};

export default PhaseBar;
