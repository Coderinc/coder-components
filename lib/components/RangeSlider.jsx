import React from 'react';
import PropTypes from 'prop-types';
import RCSlider from 'rc-slider';
import { withTheme } from 'styled-components';

const rangeSliderStyle = theme => ({
  railStyle: {
    backgroundColor: theme.offsetBackground,
  },
  dotStyle: {
    display: 'none',
  },
  activeDotStyle: {
    backgroundColor: theme.primary,
    borderColor: theme.primaryBackground,
    borderWidth: 2,
  },
  trackStyle: [
    {
      backgroundColor: theme.primary,
    },
  ],
  handleStyle: [
    {
      backgroundColor: theme.primary,
      borderColor: theme.primaryBackground,
    },
    {
      backgroundColor: theme.primary,
      borderColor: theme.primaryBackground,
    },
  ],
});

const markStyle = (theme, index, active) => {
  if (index !== 0 && index !== 10 && !active) {
    return { display: 'none' };
  }
  return {
    color: active ? theme.primaryFont : theme.offsetFont,
  };
};

class RangeSlider extends React.Component {
  constructor(props) {
    super(props);

    const { minValue = 0, maxValue = 10 } = props;
    this.marks = {};
    const delta = Math.round((maxValue - minValue) / 10);
    for (let i = 0; i < 11; i += 1) {
      const val = minValue + i * delta;
      this.marks[val] = {
        label: `${val}`,
        style: markStyle(props.theme, i, false),
      };
    }
  }

  render() {
    const {
      minValue, maxValue, input, theme,
    } = this.props;

    const marks = {};
    Object.keys(this.marks).forEach((k, i) => {
      marks[k] = {
        ...this.marks[k],
        style: markStyle(theme, i, input.value.includes(+k)),
      };
    });

    return (
      <RCSlider.Range
        defaultValue={input.value || []}
        marks={marks}
        max={maxValue}
        min={minValue}
        onChange={input.onChange}
        step={null}
        style={{ margin: '1rem 0' }}
        value={input.value}
        {...rangeSliderStyle(theme)}
      />
    );
  }
}

RangeSlider.propTypes = {
  minValue: PropTypes.number,
  maxValue: PropTypes.number,
  input: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
};

RangeSlider.defaultProps = {
  minValue: 0,
  maxValue: 10,
};

export default withTheme(RangeSlider);
