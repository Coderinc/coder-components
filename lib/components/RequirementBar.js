import React from 'react';
import styled from 'styled-components';
import Paragraph from '../elements/Paragraph';
import Chip from '../elements/Chip';

const BarContainer = styled.div`
  background-color: ${props =>
    props.active ? props.theme.activeBackground : props.theme.offsetBackground};
  color: ${props => props.theme.primaryFont};

  box-sizing: border-box;
  border-radius: 0.125rem;
  padding: 0.5rem;
  margin: 0.25rem 0;
  width: 100%;

  display: flex;

  &::before {
    font-size: 1.5rem;
    font-family: coder-icons;
    font-weight: normal;
    content: '\f108';
    margin-right: 0.5rem;
  }
`;

const BarContent = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
`;

const BarTop = styled.div`
  display: flex;
`;

const BarBottom = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

const Icon = styled.div`
  margin: 0 0.25rem;
  &::before {
    font-size: 1.5rem;
    color: ${props => props.theme.offsetFont};
    font-family: coder-icons;
    font-weight: normal;
    content: '${props => props.content}';
  }
`;

const Fill = styled.div`
  flex: 1;
`;

export const RequirementBar = ({ phases = [], label, ...props }) => (
  <BarContainer {...props}>
    <BarContent>
      <BarTop>
        <Paragraph>{label}</Paragraph>
        <Fill />
        <Icon content="\f107" />
        <Icon content="\f11f" />
      </BarTop>
      <BarBottom>{phases.map(p => (<Chip
        key={p}
        label={p}
      />))}</BarBottom>
    </BarContent>
  </BarContainer>
);

export default RequirementBar;
