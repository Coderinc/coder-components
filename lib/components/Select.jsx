import React from 'react';
import PropTypes from 'prop-types';
import Select, { components as SelectComponents } from 'react-select';
import styled, { withTheme } from 'styled-components';

export const DropdownContainer = styled.div`
  position: relative;
  height: 1.75rem;
  width: 1rem;

  span {
    position: absolute;
    display: flex;
    align-items: flex-start;
    color: ${({ theme }) => theme.offsetFont};
    font-family: coder-icons;
    height: 1rem;
    width: 1rem;
    bottom: 0;

    &::before {
      content: '\f104';
    }

    &:first-of-type {
      top: 0;
      bottom: auto;
      transform: scaleY(-1);
    }
  }
`;

const DropdownIcon = () => (
  <DropdownContainer>
    <span />
    <span />
  </DropdownContainer>
);

const DropdownIndicator = props => (
  <SelectComponents.DropdownIndicator {...props}>
    <DropdownIcon />
  </SelectComponents.DropdownIndicator>
);

const SingleValueContainer = styled.div`
  display: flex;
  align-items: center;

  span {
    text-transform: capitalize;
    max-width: 100%;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  img {
    width: 1rem;
    height: 1rem;
    margin-right: 0.5rem;
  }
`;

const SingleValue = ({ data, ...props }) => (
  <SelectComponents.SingleValue {...props}>
    <SingleValueContainer>
      {data.icon && <img alt="Icon" src={data.icon} />}
      <span>
        {data.label}
      </span>
    </SingleValueContainer>
  </SelectComponents.SingleValue>
);

SingleValue.propTypes = {
  data: PropTypes.object.isRequired,
};

const PlaceholderContainer = styled.div`
  display: flex;
  align-items: center;

  span {
    color: ${({ theme }) => theme.offsetFont};
  }

  img {
    width: 1rem;
    height: 1rem;
    margin-right: 0.5rem;
  }
`;

const Placeholder = ({ icon, label, ...props }) => (
  <SelectComponents.Placeholder {...props}>
    <PlaceholderContainer>
      {icon && <img alt="Icon" src={icon} />}
      <span>
        {label || 'Select...'}
      </span>
    </PlaceholderContainer>
  </SelectComponents.Placeholder>
);

Placeholder.propTypes = {
  icon: PropTypes.string,
  label: PropTypes.string,
};

Placeholder.defaultProps = {
  icon: undefined,
  label: '',
};

const OptionContainer = styled.div`
  display: flex;
  align-items: center;

  span {
    color: ${({ theme }) => theme.offsetFont};
  }

  img {
    width: 1rem;
    height: 1rem;
    margin-right: 0.5rem;
  }
`;

const Option = ({ data, ...props }) => (
  <SelectComponents.Option {...props}>
    <OptionContainer>
      {data.icon && <img alt="Icon" src={data.icon} />}
      <span>
        {data.label}
      </span>
    </OptionContainer>
  </SelectComponents.Option>
);

Option.propTypes = {
  data: PropTypes.object.isRequired,
};

const MultiValueLabelContainer = styled.div`
  display: flex;
  align-items: center;
`;

const MultiValueLabel = ({ icon, label, ...props }) => (
  <SelectComponents.MultiValueLabel {...props}>
    <MultiValueLabelContainer>
      {icon && <img alt="Icon" src={icon} />}
      <span>
        {label}
      </span>
    </MultiValueLabelContainer>
  </SelectComponents.MultiValueLabel>
);

MultiValueLabel.propTypes = {
  icon: PropTypes.string,
  label: PropTypes.string,
};

MultiValueLabel.defaultProps = {
  icon: undefined,
  label: '',
};

const MultiValue = ({ data, ...props }) => (
  <SelectComponents.MultiValue {...props}>
    <MultiValueLabel {...data} />
  </SelectComponents.MultiValue>
);

MultiValue.propTypes = {
  data: PropTypes.object.isRequired,
};

const themer = theme => ({
  container: base => ({
    ...base,
    letterSpacing: '0.0625rem',
    fontWeight: '200',
    fontSize: '0.875rem',
    color: theme.primaryFont,
    width: '100%',
  }),
  control: (base, state) => ({
    ...base,
    backgroundColor: state.isFocused ? theme.offsetBackground : theme.primaryBackground,
    width: '100%',
    border: 'none',
    boxShadow: 'none',
    cursor: 'text',
    transition: 'none',
  }),
  clearIndicator: base => ({
    ...base,
    color: theme.offsetFont,
    cursor: 'pointer',
    '&:hover': {
      color: theme.primary,
    },
  }),
  singleValue: base => ({
    ...base,
    color: theme.fontColor,
  }),
  multiValue: base => ({
    ...base,
    backgroundColor: theme.activeBackground,
    color: theme.primaryFont,
    img: {
      width: '0.75rem',
      height: '0.75rem',
      marginRight: '0.25rem',
    },
  }),
  multiValueLabel: base => ({
    ...base,
    color: theme.primaryFont,
    textTransform: 'capitalize',
  }),
  multiValueRemove: base => ({
    ...base,
    cursor: 'pointer',
    '&:hover': {
      color: theme.primary,
      backgroundColor: theme.activeBackground,
    },
  }),
  placeholder: base => ({
    ...base,
    color: theme.offsetFont,
  }),
  input: base => ({
    ...base,
    color: theme.primaryFont,
  }),
  menu: base => ({
    ...base,
    backgroundColor: theme.offsetBackground,
    // boxShadow: 'none',
    boxShadow: '0 14px 12px rgba(0,0,0,.25), 0 10px 8px rgba(0,0,0,.22)',
    borderRadius: 0,
    marginTop: '-0.25rem',
  }),
  option: (base, state) => ({
    ...base,
    '&:hover': {
      color: theme.primary,
      backgroundColor: theme.activeBackground,
    },
    color: state.isSelected ? theme.primary : theme.primaryFont,
    backgroundColor: 'transparent',
    cursor: 'pointer',
  }),
  indicatorSeparator: base => ({
    ...base,
    backgroundColor: 'none',
  }),
  dropdownIndicator: base => ({
    ...base,
    cursor: 'pointer',
    color: theme.primaryFont,
    '&:hover': {
      color: theme.offsetFont,
    },
  }),
});

export const ThemelessSingleSelect = ({ theme, ...props }) => (
  <Select
    {...props}
    components={{
      DropdownIndicator,
      SingleValue,
      Option,
      Placeholder,
    }}
    styles={themer(theme)}
  />
);

ThemelessSingleSelect.propTypes = {
  theme: PropTypes.object.isRequired,
};

export const ThemelessMultiSelect = ({ theme, ...props }) => (
  <Select
    {...props}
    components={{
      DropdownIndicator,
      Option,
      MultiValue,
    }}
    isMulti
    styles={themer(theme)}
  />
);

ThemelessMultiSelect.propTypes = {
  theme: PropTypes.object.isRequired,
};

export const SingleSelect = withTheme(ThemelessSingleSelect);
export const MultiSelect = withTheme(ThemelessMultiSelect);
