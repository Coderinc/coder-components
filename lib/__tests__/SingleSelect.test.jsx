import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { SingleSelect } from '../components/Select';

const options = [
  { value: 1, label: 'A' },
  { value: 2, label: 'B' },
];

describe('SingleSelect', () => {
  it('renders correctly', () => {
    const component = renderer.create(<SingleSelect options={options} theme={{}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot(); 
  });
});