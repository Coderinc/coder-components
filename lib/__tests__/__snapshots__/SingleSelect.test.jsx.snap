// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleSelect renders correctly 1`] = `
.c1 {
  position: relative;
  height: 1.75rem;
  width: 1rem;
}

.c1 span {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  font-family: coder-icons;
  height: 1rem;
  width: 1rem;
  bottom: 0;
}

.c1 span::before {
  content: '\\f104';
}

.c1 span:first-of-type {
  top: 0;
  bottom: auto;
  -webkit-transform: scaleY(-1);
  -ms-transform: scaleY(-1);
  transform: scaleY(-1);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 img {
  width: 1rem;
  height: 1rem;
  margin-right: 0.5rem;
}

<div
  className="css-ohrm6p"
  onKeyDown={[Function]}
>
  <div
    className="css-1dd3glh"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="css-1rtrksz"
    >
      <div
        className="css-10tzkad"
      >
        <div
          className="c0"
        >
          <span>
            Select...
          </span>
        </div>
      </div>
      <div
        className="css-3y1qaa"
      >
        <div
          className=""
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-autocomplete="list"
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            disabled={false}
            id="react-select-2-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            spellCheck="false"
            style={
              Object {
                "background": 0,
                "border": 0,
                "boxSizing": "content-box",
                "color": "inherit",
                "fontSize": "inherit",
                "opacity": 1,
                "outline": 0,
                "padding": 0,
                "width": "1px",
              }
            }
            tabIndex="0"
            type="text"
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-1wy0on6"
    >
      <span
        className="css-1xxsjqy"
      />
      <div
        aria-hidden="true"
        className="css-8gakrk"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className="c1"
        >
          <span />
          <span />
        </div>
      </div>
    </div>
  </div>
</div>
`;
